docs/deployment/debian_ubuntu.md

```diff
@@ -0,0 +1,236 @@
+```
+
+### 3) Make the starter script
+
+This step is recommended for two reasons: first, to reduce the complexity of the desktop shortcut, and second, to make your app runnable from Terminal.
+
+The starter script file name should preferrably be `your_program` (without `.sh` extension), so whenever your user types "your_program" on the Terminal, he / she will start your program.
+
+To run from command-line, we need to append the libs directory path to the **LD_LIBRARY_PATH** environment variable - this is to make native libraries, such as `libHarfBuzzSharp.so` and `libSkiaSharp.so`, available to your program during initialization.
+
+:::warning
+
+Without this, your program will crash due to missing libraries!
+
+:::
+
+Example of `your_program` starter script, setting the variable in a command-scope:
+
+```sh
+#!/bin/bash
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/your_program your_program_executable
+```
+
+### 4) Make the desktop shortcut
+
+The desktop shortcut file follows the [freedesktop specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys). Arch Linux Wiki also has [good related information](https://wiki.archlinux.org/title/Desktop_entries).
+
+Below is an example of a desktop shortcut file:
+
+```
+[Desktop Entry]
+Name=YourProgram
+Comment=YourProgram, great for doing X
+Icon=your_program
+Exec=your_program
+Terminal=false
+Type=Application
+Categories=Development
+GenericName=YourProgram
+Keywords=keyword1; keyword2; keyword3
+```
+
+### 5) Add hicolor icons (optional)
+
+Hicolor icons follow a folder structure like below.
+
+[This blog post](https://martin.hoppenheit.info/blog/2016/where-to-put-application-icons-on-linux/) advises us to put icons on both `hicolor` and `pixmaps` directories, according to [Debian Menu System docs](https://www.debian.org/doc/packaging-manuals/menu.html/ch3.html#s3.7) and [FreeDesktop docs](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-0.11.html#install_icons).
+
+```
+├── icons
+│   └── hicolor
+│       ├── 128x128
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 16x16
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 256x256
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 32x32
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 48x48
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 512x512
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 64x64
+│       │   └── apps
+│       │       └── your_program.png
+│       └── scalable
+│           └── apps
+│               └── your_program.svg
+```
+
+### 6) Compile the `.deb` package
+
+```sh
+# for x64 architectures, the suggested suffix is amd64.
+dpkg-deb --root-owner-group --build "./staging_folder/" "./your_program_${versionName}_amd64.deb"
+```
+
+## Example of a PowerShell script for the entire process
+
+```powershell
```

Comment:
I know powershell works on Linux but I don't think I have ever seen anybody actually use it. Is it common in packaging scenarios?

Reply:
Changed into Linux shell script, as it makes more sense to Linux users